test/ec_comp_test.c

```diff
@@ -0,0 +1,2310 @@
+        return 1;
+    if (known_curves[curve_i].nid == nid_Oakley_EC2N_4)
+        return 1;
+    return 0;
+}
+
+#ifndef GENERATE_KNOWN_RESULTS
+static int verify_known_result(size_t curve_i, size_t param_format_i, size_t comp_format_i, size_t key_src, struct default_config_st *dcfg)
+{
+    size_t i;
+
+    for (i=0; i<OSSL_NELEM(known_results); ++i) {
+        if (strcmp(known_results[i].curve_name, known_curve_names[curve_i]) != 0)
+            continue;
+        if (param_formats_i[param_format_i] != known_results[i].param_format)
+            continue;
+        if (comp_formats_i[comp_format_i] != known_results[i].comp_format)
+            continue;
+        if (key_src != known_results[i].key_src)
+            continue;
+
+        break;
+    }
+
+    if (i >= OSSL_NELEM(known_results)) {
+        fprintf(stderr, "# could not identify known result\n");
+        return 0;
+    }
+
+    if (known_results[i].res_param_format != dcfg->asn1_flag) {
+        fprintf(stderr, "# EC parameter format mismatch\n");
+        return 0;
+    }
+
+    if (known_results[i].res_comp_format != dcfg->point_form) {
+        fprintf(stderr, "# EC compression format mismatch\n");
+        return 0;
+    }
+
+    return 1;
+}
+#endif
+
+/*
+ * curve_i: Index into known_curves.
+ * param_format_i: 0=Named parameters, 1=Explicit parameters.
+ * comp_format_i: Index into comp_formats(_i).
+ *   If this exceeds the size of comp_formats, does not configure compression
+ *   format.
+ * key_src: 0=Deserialize an existing key. 1=Generate a new key.
+ */
+static int comp_test_actual(size_t curve_i,
+                            size_t param_format_i,
+                            size_t comp_format_i,
+                            size_t key_src)
+{
+    int rc = 0;
+    EC_KEY *k = NULL;
+    unsigned char *buf = NULL;
+    int buf_len;
+    struct default_config_st dcfg = {0};
+    char curve_name[256];
+
+    if (should_skip(curve_i, param_format_i))
+        return 1;
+
+    snprintf(curve_name, sizeof(curve_name), "\"%s\"", known_curve_names[curve_i]);
+
+#ifdef GENERATE_KNOWN_RESULTS
+    fprintf(stderr, "# KNOWN_RESULT(%26s,%16s,%16s,%9s,",
+        curve_name, param_formats[param_format_i],
+        comp_formats[comp_format_i], key_src ? "generate" : "existing");
+#endif
+
+    k = (key_src == 0)
+        ? get_existing_key(curve_i,
+                           param_format_i,
+                           comp_format_i,
+                           &dcfg)
+        : generate_new_key(curve_i,
+                           param_formats_i[param_format_i],
+                           comp_formats_i[comp_format_i],
+                           &dcfg);
+    if (k == NULL)
+        return 0;
+
+#ifdef GENERATE_KNOWN_RESULTS
+    fprintf(stderr, "    %11s, %8s)\\\n", decode_asn1_flag(dcfg.asn1_flag), decode_point_form(dcfg.point_form));
+#else
+    if (verify_known_result(curve_i, param_format_i, comp_format_i, key_src, &dcfg) == 0)
+        goto fail;
+#endif
```

Comment:
I am not sure in this case putting the verification steps in a separate function adds to readability, so in the other comments I am assuming all the verification steps are done in this same function, rather than calling `verify_known_result`.